### PR TITLE
Fixture for initializing the microraiden.requests module with a tester-aware DefaultHTTPClient

### PR DESCRIPTION
**Pull request by devium**
`microraiden.requests` could not be used easily in tests, since there was no fixture for it. The new `init_microraiden_requests` fixture simply calls a new variant of `microraiden.requests.init()` with the existing `default_http_client` fixture.

New µRaiden client tests can now be written like this:
```
def test_something(init_microraiden_requests, some_microraiden_server):
    import microraiden.requests
    response = microraiden.requests.get('https://localhost:5000/that_resource')
    assert response...
```

### CODE REVIEW

File: microraiden/microraiden/requests/__init__.py [hunk 1]
```diff
@@ -3,15 +3,16 @@
...
 from microraiden.contract_proxy import ChannelContractProxy, ContractProxy
 
-client = None
-http_client = None
+_http_client = None
 
 
 def init(
+    http_client: HTTPClient = None,
+    client_: Client = None,
```
Comment (pcppcp) [inline microraiden/microraiden/requests/__init__.py]: underscore at the end is ugly

File: microraiden/microraiden/requests/__init__.py [hunk 2]
```diff
@@ -3,15 +3,16 @@
...
 from web3 import Web3
 
-from microraiden import DefaultHTTPClient, Client
+from microraiden import DefaultHTTPClient, Client, HTTPClient
 from microraiden.config import CHANNEL_MANAGER_ADDRESS, CONTRACT_METADATA
 from microraiden.contract_proxy import ChannelContractProxy, ContractProxy
 
-client = None
-http_client = None
+_http_client = None
```
Comment (pcppcp) [inline microraiden/microraiden/requests/__init__.py]: So it's not possible to use two different keys in one session? It's probably ok to have a global now, but the library should also support creating multiple `request` contexts.

Comment (devium) [reply microraiden/microraiden/requests/__init__.py]: The use of multiple clients/keys is planned for #256 and will follow the design of `requests.Session`.
Currently if you want multiple clients you just have to set up your own `DefaultHTTPClient`s again which is pretty much the same as calling `init()` on the global client.

File: microraiden/microraiden/requests/__init__.py (hunk 1 above)
Comment (devium) [reply microraiden/microraiden/requests/__init__.py]: Agreed, will fix. Looks like an earlier version.

**Pull request merged**